subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc

```diff
@@ -120,6 +120,11 @@ A Gradle build is defined by a `settings.gradle(.kts)` file in the current or pa
 Without a settings file, a Gradle build is undefined.
 Executing Gradle in such a directory is now an error.
 
+==== Calling Project.afterEvaluate() after project evaluation is now an error
+
+Gradle 6.x warns users about the wrong behavior and ignores the target action in this scenario.
+Starting from 7.0 the same case will produce an error. Plugins and build scripts should be adjusted to call `afterEvaluate` only at configuration time.
```

Comment:
Is there any general advice to give people encountering this? What did happen before? Would the block have been ignored anyway? If possible, I think something actionable would be great to add here

Reply:
The `afterEvaluate` we set to fail now was ignored in Gradle versions <7.0. How about this:

_If you have such a build failure and the related `afterEvaluate` statement is declared in your build sources then you can simply delete it. If `afterEvaluate` is declared in a plugin then report the issue to the plugin maintainers._

Reply:
Perfect!